mm2src/coins/lightning_persister/src/lib.rs

```diff
@@ -278,38 +279,48 @@ fn payment_info_from_row(row: &Row<'_>) -> Result<PaymentInfo, SqlError> {
     let is_outbound = row.get::<_, bool>(8)?;
     let payment_type = if is_outbound {
         PaymentType::OutboundPayment {
-            destination: PublicKey::from_str(&row.get::<_, String>(1)?).expect("PublicKey from str should not fail!"),
+            destination: PublicKey::from_str(&row.get::<_, String>(1)?)
+                .map_err(|e| SqlError::FromSqlConversionFailure(1, SqlType::Text, Box::new(e)))?,
```

Comment:
Creating such error is repeating in this file. I have several ideas how it can be simplified:
1) Return a custom error type:
```rust
enum MmSqlError {
  ErrorParsing {
     // Both or one of these:
    field_name: String,
    field_id: usize,
    error: String,
  }
}

impl LnStorageError {
  pub fn error_parsing<E>(field_name: String, field_id: usize, error: E) -> Self where E: Error
  {
    MmSqlError::ErrorParsing {field_name, field_id, error: error.to_string()}
  }
}
```
2) Use a function like:
```rust
pub fn sql_text_conversion_err<E>(field_id: usize, e: E) -> SqlError {
  SqlError::FromSqlConversionFailure(field_id, SqlType::Text, Box::new(e))
}
```
What do you think?

Reply:
> ```
> pub fn sql_text_conversion_err<E>(field_id: usize, e: E) -> SqlError {
>   SqlError::FromSqlConversionFailure(field_id, SqlType::Text, Box::new(e))
> }
> ```

I prefer this idea since this function can be used easily for any such error. I will add it to our db_common lib.